### PR TITLE
Add vm to TestAuthorization_Deny and TestAuthorization_NegativeMatch

### PR DESCRIPTION
**Pull request by JimmyCYJ**
This splits out of #31618

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[x] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: tests/integration/security/authorization_test.go [hunk 1]
```diff
@@ -361,22 +355,22 @@ func TestAuthorization_Deny(t *testing.T) {
...
-						newTestCase(b, "/allow?param=value", true),
-						newTestCase(c, "/allow/admin", false),
-						newTestCase(c, "/allow/admin?param=value", false),
-						newTestCase(c, "/global-deny", false),
-						newTestCase(c, "/global-deny?param=value", false),
-						newTestCase(c, "/other", false),
-						newTestCase(c, "/other?param=value", false),
-						newTestCase(c, "/allow", true),
-						newTestCase(c, "/allow?param=value", true),
+						newTestCase(dst0, "/deny", false),
```
Comment (yangminzhu) [inline tests/integration/security/authorization_test.go:358]: I think this still reduce the current test coverage:

In the old test, we tested all the 16 cases on k8s workload (even they are different but it's both k8s).

But in the new test, we only test the first 8 cases on k8s workload and then the next 8 cases on the vm workload. This means the first 8 cases are not tested on vm workload, and the last 8 cases are not tested on k8s workload.

Could we make the test to run all cases on both platform:
- Test k8s workload b and c (current case)
- Test vm workload vm1 and vm2 (new test, just replace b with vm1 and c with vm2)

File: tests/integration/security/authorization_test.go [hunk 2]
```diff
@@ -392,23 +386,21 @@ func TestAuthorization_NegativeMatch(t *testing.T) {
 		Run(func(t framework.TestContext) {
 			ns := apps.Namespace1
 			ns2 := apps.Namespace2
-
+			dst0 := apps.VM.Match(echo.Namespace(apps.Namespace1.Name()))
```
Comment (yangminzhu) [inline tests/integration/security/authorization_test.go:389]: I think this has the similar reduced test coverage issue like I mentioned above, please see my other comment.

File: tests/integration/security/authorization_test.go (hunk 1 above)
Comment (JimmyCYJ) [reply tests/integration/security/authorization_test.go:358]: I have reverted back for b and c, the policies are still templatized.
I also added more policies for vm, the policy is similar to c. This is not perfect, but I think it is good as an initial step. Now we have test policy templatized, and vm included. How to revise the policies for better coverage for vm could be left for future work. What do you think?

File: tests/integration/security/authorization_test.go (hunk 2 above)
Comment (JimmyCYJ) [reply tests/integration/security/authorization_test.go:389]: I have reverted the tests back to b,c,d and add more for VM. I also add a TODO to revise the vm tests with multi-vm support and test updates.

File: tests/integration/security/testdata/authz/v1beta1-negative-match.yaml.tmpl [hunk 3]
```diff
@@ -22,38 +22,57 @@ spec:
...
-      "app": "d"
+      "app": "{{ .dst2 }}"
   action: DENY
   rules:
   - from:
     - source:
         notPrincipals: ["*"]
 ---
 
+# The following policy denies access to path with prefix "/prefix" except "/prefix/allowlist" to workload
```
Comment (yangminzhu) [inline tests/integration/security/testdata/authz/v1beta1-negative-match.yaml.tmpl:57]: can you also add a TODO that the following policy is a duplicate of `policy-{{ .dst0 }}-deny` and can be removed once the test framework supports multiple vm workloads?

File: tests/integration/security/testdata/authz/v1beta1-deny.yaml.tmpl [hunk 4]
```diff
@@ -1,50 +1,86 @@
...
-      "app": "c"
+      "app": "{{ .dst1 }}"
+  action: ALLOW
+  rules:
+  - to:
+    - operation:
+        paths: ["/allow*"]
+---
+
+# The following policy denies access to path /allow/admin to workload
```
Comment (yangminzhu) [inline tests/integration/security/testdata/authz/v1beta1-deny.yaml.tmpl:55]: Same here, can you add a comment that the following policies are duplicated from existing ones and can be removed if the test framework supports multiple vm workloads.

File: tests/integration/security/authorization_test.go (hunk 1 above)
Comment (yangminzhu) [reply tests/integration/security/authorization_test.go:360]: Sounds good, we can always add more tests later but we should not reduce the test coverage of current tests.

**Review by yangminzhu — APPROVED** [review]
can you open an issue for supporting multiple vm workloads in the test framework? It's needed by multiple security tests.

File: tests/integration/security/testdata/authz/v1beta1-negative-match.yaml.tmpl (hunk 3 above)
Comment (JimmyCYJ) [reply tests/integration/security/testdata/authz/v1beta1-negative-match.yaml.tmpl:57]: Thanks, done.

File: tests/integration/security/testdata/authz/v1beta1-deny.yaml.tmpl (hunk 4 above)
Comment (JimmyCYJ) [reply tests/integration/security/testdata/authz/v1beta1-deny.yaml.tmpl:55]: Added TODO here. Thanks.

**Comment by JimmyCYJ** [comment]
> can you open an issue for supporting multiple vm workloads in the test framework? It's needed by multiple security tests.
> 
> lgtm, left a few small comments, feel free to remove the hold label after addressing.

Thanks! I will create an issue for multi-vm support. @stevenctl

**Pull request merged**